includes/admin/class-wc-admin-notices.php

```diff
@@ -530,6 +546,32 @@ public static function simplify_commerce_notice() {
 	public static function theme_check_notice() {
 		wc_deprecated_function( 'WC_Admin_Notices::theme_check_notice', '3.3.0' );
 	}
+
+	/**
+	 * Check if uploads directory is public.
+	 *
+	 * @since 4.2.0
+	 * @return bool
+	 */
+	protected static function is_uploads_directory_is_public() {
+		$uploads = wp_upload_dir( null, true );
+
+		// Skip if returns an error.
+		if ( $uploads['error'] ) {
+			return true;
+		}
+
+		// Check for uploads root.
+		$baseurl_response = wp_safe_remote_get( $uploads['baseurl'] );
```

Comment:
We should add some caching around this to avoid running the HTTP requests on every page load.